crypto/keyspb/keyspb.proto

```diff
@@ -0,0 +1,63 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package keyspb;
+
+// Generator defines how a new pair of asymmetric keys should be generated.
+message Generator {
+  message ECDSA {
+    enum Curve {
+      P256 = 0;
+      P224 = 1;
+      P384 = 2;
+      P521 = 3;
+    }
+    Curve curve = 1;
+  }
+
+  message RSA {
+    // Size of the keys in bits.
```

Comment:
Go's crypto/rsa package doesn't document any constraints but, looking at its implementation, the only constraint appears to be that `bits` must be large enough to generate 2 primes. I'll document that 0 is treated as "use a default key size" too.